File: Main/static/JS/game/effects/solidity.js
import { SummaryDurableEffect } from "./effect.js";
import { Modificator } from "../classes/modificator.js";

class Solidity extends SummaryDurableEffect{
    constructor(hero, game_state, duration, caster){
        super('стойкость', 'solidity.png', hero, game_state, duration, caster)
        this.description = 'Герой игнорирует весь урон от первого удара по нему.'
    }

    gen_description(){
        return `Стойкость[${this.duration}] (наложено ${this.caster.name})`
    }

    before_move(){
        this.decrease_duration()
    }

    gen_modificators(){
        return [new Modificator('solidity', 1, true, 'after', 'set')]
    }
}

export {Solidity}

File: Main/models/effects/solidity.py
from .effect import SummaryDurableEffect
from django.db import models
from Main.models.modifier import Modifier


class Solidity(SummaryDurableEffect):
    name = 'Solidity'
    img_src = 'solidity.png'

    def gen_description(self, link):
        return f"Стойкость ({self.duration} ходов) (наложено {link.caster.name})"

    def before_move(self, link):
        self.decrease_duration(link)

    def gen_modifiers(self, link):
        return [Modifier(hero=link.hero, param_name='solidity', value=True, duration=1, is_set=True)]
